src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -270,6 +270,28 @@ def _process_resource_change(
         ]
 
         for installation in installations:
+            try:
+                service_hook = ServiceHook.objects.get(
+                    organization_id=org.id,
+                    actor_id=installation.id,
+                )
+            except ServiceHook.DoesNotExist:
+                logger.info(
+                    "send_webhooks.missing_servicehook",
+                    extra={"installation_id": installation.id, "event": event},
+                )
+                continue
+
+            # Check if service hook is limited to specific projects.
+            # (applies to all projects if there are no ServiceHookProject records)
+            if ServiceHookProject.objects.filter(service_hook_id=service_hook.id).exists():
+                project_allowed = ServiceHookProject.objects.filter(
+                    service_hook_id=service_hook.id, project_id=instance.project_id
+                ).exists()
```

Comment:
Are we trying to do some feature flag here/exclude other orgs without an explicit feature flag? I'm confused why there's two lookups, 1 for if the `ServiceHook` has any attached `ServiceHookProjects` and then another check for if there exists a `ServiceHookProject` for the processed event's project. Like can we combine the two?

Reply:
Yes, no records means all projects receive webhooks 
but if there's at least one "filter" (`exists()`) then only projects for which records exist receive webhook

Not sure that's possible to capture in a single expression with Django model manager

Reply:
ohh I see, since currently `SentryAppInstallations` only create `servicehooks` without any `servicehookprojects` (i.e no per project filtering), we're using `ServiceHookProject` as a flag to see if this app has per project filtering, if they do then get and apply filters. makes sense to me

Reply:
yep exactly